index.js

```diff
@@ -214,23 +241,23 @@ class Markdown extends Component {
 
 
         switch (node.type) {
-            case 'h1': return this.renderText(node, key, Utils.concatStyles(extras, styles.h1));
-            case 'h2': return this.renderText(node, key, Utils.concatStyles(extras, styles.h2));
-            case 'h3': return this.renderText(node, key, Utils.concatStyles(extras, styles.h3));
-            case 'h4': return this.renderText(node, key, Utils.concatStyles(extras, styles.h4))
-            case 'h5': return this.renderText(node, key, Utils.concatStyles(extras, styles.h5));
-            case 'h6': return this.renderText(node, key, Utils.concatStyles(extras, styles.h6));
+            case 'h1': return this.renderText(node, key, Utils.concatStyles(extras, styles.h1), 'h1' );
+            case 'h2': return this.renderText(node, key, Utils.concatStyles(extras, styles.h2), 'h2' );
+            case 'h3': return this.renderText(node, key, Utils.concatStyles(extras, styles.h3), 'h3' );
+            case 'h4': return this.renderText(node, key, Utils.concatStyles(extras, styles.h4), 'h4' );
+            case 'h5': return this.renderText(node, key, Utils.concatStyles(extras, styles.h5), 'h5' );
+            case 'h6': return this.renderText(node, key, Utils.concatStyles(extras, styles.h6), 'h6' );
             case 'hr': return this.renderLine(node, key);
             case 'div': return this.renderBlock(node, key, extras);
             case 'ul': return this.renderList(node, key, false);
             case 'ol': return this.renderList(node, key, true);
             case 'li': return this.renderListItem(node, key, index, extras);
             case 'a': return this.renderLink(node, key);
             case 'img': return this.renderImage(node, key);
-            case 'strong': return this.renderText(node, key, Utils.concatStyles(extras, styles.strong));
-            case 'del': return this.renderText(node, key, Utils.concatStyles(extras, styles.del));
-            case 'em': return this.renderText(node, key, Utils.concatStyles(extras, styles.em));
-            case 'u': return this.renderText(node, key, Utils.concatStyles(extras, styles.u));
+            case 'strong': return this.renderText(node, key, Utils.concatStyles(extras, styles.strong), 'strong');
+            case 'del': return this.renderText(node, key, Utils.concatStyles(extras, styles.del), 'del');
+            case 'em': return this.renderText(node, key, Utils.concatStyles(extras, styles.em), 'em');
+            case 'u': return this.renderText(node, key, Utils.concatStyles(extras, styles.u), 'u');
```

Comment:
Let renderText know the node type, so that we can pass it to a consumer of `renderText`

Reply:
Would be great to add renderHTML functions to customize html code, if there is any. In what I am trying, markdown usually has some sort of html codes e.g. <center>, <br>, <table>, etc. etc.

Reply:
Are we missing support for some markdown syntax?

https://www.markdownguide.org/basic-syntax/

This PR was trying to expose our internal renderers to allow you to customize the rendering of these tags rather than only overriding our default styling.

But when implementing it elsewhere in our codebase, it was pretty tricky to get right. 

Not sure if this would be good to put in master, as implementing your own renderers was painful. i.e `Views` being rendered inside a block quote that already opened a `Text` element will break.

Reply:
Tables are missing, but I see them in roadmap, also possibility to customize some common html tags would be very useful

> Are we missing support for some markdown syntax?
> 
> https://www.markdownguide.org/basic-syntax/
> 
> This PR was trying to expose our internal renderers to allow you to customize the rendering of these tags rather than only overriding our default styling.
> 
> But when implementing it elsewhere in our codebase, it was pretty tricky to get right.
> 
> Not sure if this would be good to put in master, as implementing your own renderers was painful. i.e `Views` being rendered inside a block quote that already opened a `Text` element will break.